.github/workflows/build-pr.yml

```diff
@@ -27,6 +27,9 @@ jobs:
             exit 1
           fi
 
+      - name: Install xmllint
+        uses: ./.github/actions/install-xmllint
+
```

Comment:
Maybe we should use SNAPSHOT artifacts in the build-pr workflow to catch any problems early

Reply:
Also: How this will work with backports to the maintainance branches?

Reply:
> Maybe we should use SNAPSHOT artifacts in the build-pr workflow to catch any problems early

What problems are you envisaging?

The risk of testing `SNAPSHOT`s against PRs is that there could be an breaking change on the `hazelcast-mono` side that would only be visible with an unrelated change on the `hazelcast-docker` side.

> Also: How this will work with backports to the maintainance branches?

not sure what you mean specifically - do you mean the maintenance branches of the Docker images?

I'd guess that older versions (>`5.4`) would be fine as the artefacts are still public, but `5.4.z` probably will have the same issue. So I guess we'd need to backport this there also? That could be unpleasant.

Reply:
> > Maybe we should use SNAPSHOT artifacts in the build-pr workflow to catch any problems early
> 
> What problems are you envisaging?

- a new native library required to be installed by apk/microdnf
- incompatible changes related to
   - `hazelcast-download.properties`
   -  log4j2.properties log4j2-json.properties 
   -  jmx_agent_config.yaml
   - then change of start command (`hz start`)


> The risk of testing `SNAPSHOT`s against PRs is that there could be an breaking change on the `hazelcast-mono` side that would only be visible with an unrelated change on the `hazelcast-docker` side.

I would prefrer failing fast during the development rather than finding out about the issue very late during the release pipeline

> > Also: How this will work with backports to the maintainance branches?
> not sure what you mean specifically - do you mean the maintenance branches of the Docker images?

yes, I mean `5.4.z` branch

Reply:
> > > Maybe we should use SNAPSHOT artifacts in the build-pr workflow to catch any problems early
> > 
> > 
> > What problems are you envisaging?
> 
> * a new native library required to be installed by apk/microdnf
> * incompatible changes related to
>   
>   * `hazelcast-download.properties`
>   * log4j2.properties log4j2-json.properties
>   * jmx_agent_config.yaml
>   * then change of start command (`hz start`)
> 
> > The risk of testing `SNAPSHOT`s against PRs is that there could be an breaking change on the `hazelcast-mono` side that would only be visible with an unrelated change on the `hazelcast-docker` side.
> 
> I would prefrer failing fast during the development rather than finding out about the issue very late during the release pipeline

Ok - I _think_ this is a separate requirement as we don't have that already?

We only have coverage of `SNAPSHOT` Hazelcast releases when someone raises a (potentially unrelated) PR.
With this change, we won't have that either.

If we wanted proper coverage, I think we should run something _like_ the `build-pr.yml` action, but on a scheduled basis, pointing at Hazelcast's `SNAPSHOT`s?

Reply:
I think that for now it's enough to run if someone raises a (potentially unrelated) PR

Reply:
a0b2f0ce3f9219e1a4cdd528ae79b2f4163a1f6e